content/en/docs/usage/certificate.md

```diff
@@ -97,79 +96,130 @@ The `Certificate` will be issued using the issuer named `ca-issuer` in the
 > issued X.509 certificates before the issue time to fix clock-skew issues,
 > leading to the working duration of a certificate to be less than the full
 > duration of the certificate. For example, Let's Encrypt sets it to be one hour
-> before issue time, so the actual *working duration* of the certificate is 89
-> days, 23 hours (the *full duration* remains 90 days).
+> before issue time, so the actual _working duration_ of the certificate is 89
+> days, 23 hours (the _full duration_ remains 90 days).
 
 A full list of the fields supported on the Certificate resource can be found in
-the [API reference documentation](../../reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec)
+the [API reference documentation](../../reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec).
 
-## Key Usages
+## X.509 key usages and extended key usages {#key-usages}
 
-cert-manager supports requesting certificates that have a number of custom key
-usages and extended key usages. Although cert-manager will attempt to honor this
-request, some issuers will remove, add defaults, or otherwise completely ignore
-the request and is determined on an issuer by issuer basis. The `CA` and
-`SelfSigned` `Issuer` will always return certificates matching the usages you have
-requested.
+cert-manager supports requesting certificates that have a number of [custom key
+usages](https://tools.ietf.org/html/rfc5280#section-4.2.1.3) and [extended key
+usages](https://tools.ietf.org/html/rfc5280#section-4.2.1.12). Although
+cert-manager will attempt to honor this request, some issuers will remove, add
+defaults, or otherwise completely ignore the request and is determined on an
+issuer by issuer basis. The `CA` and `SelfSigned` `Issuer` will always return
+certificates matching the usages you have requested.
 
 Unless any number of usages has been set, cert-manager will set the default
-requested usages of "digital signature", "key encipherment", and "server auth".
+requested usages of `digital signature`, `key encipherment`, and `server auth`.
 cert-manager will not attempt to request a new certificate if the current
-certificate does not match the current key usages set.
+certificate does not match the current key usage set.
 
 An exhaustive list of supported key usages can be found in the [API reference
 documentation](../../reference/api-docs/#cert-manager.io/v1alpha2.KeyUsage).
 
-## Temporary Certificates whilst Issuing
+## Temporary Certificates while Issuing
+
+When requesting certificates [using the ingress-shim](../ingress/), the
+component `ingress-gce`, if used, requires that a temporary certificate is
+present while waiting for the issuance of a signed certificate when serving. To
+facilitate this, if the following annotation:
```

Comment:
I found two places where this "trick" is mentioned:

- the actual PR where this annotation was added: https://github.com/jetstack/cert-manager/pull/2089,
- In the [release notes for 0.11](https://github.com/jetstack/cert-manager/blob/dffbf391dbb0fc6c1cfea62e561a9c6f54362ab0/design/release-notes/release-0.11/draft-release-notes.md#no-longer-generating-temporary-certificates-by-default):

  > **New CertificateRequest resource type**
  >
  > The activation of `CertificateRequest` controllers are no longer behind a feature flag and are now instead enabled by default.
  >
  > This means that when requesting certificates using the `Certificate` resource the `CertificateRequest` resource will be used as the default and only way to honour the request.
  >
  > The addition of this resource introduces the ability for much greater extension points to cert-manager, notably out-of-tree issuers, istio integrations, and experimental tooling such as a CSI driver. You can read more about the motivation and design of this resource in the [enhancement document](https://github.com/jetstack/cert-manager/blob/master/design/20190708.certificate-request-crd.md).
  >
  > This change should cause no disruption to how end users interact with cert-manager, with the exception of debugging now requiring this resource to be inspected also.

- But by far the best explanation has been written by James Munnelly in https://github.com/kubernetes/ingress-gce/pull/388:

  > This switches ingress-gce back to the previous behaviour of 'soft failing' if the specified TLS secret exists. Without this change, we've had many users complain that auto-TLS using cert-manager (and also kube-lego) is broken.

I'll go ahead and add a link to this ingress-gce PR.

Reply:
I now use the past tense since this seems to be a solved issue:

> On old GKE versions (1.10.7-gke.1 and below), when requesting certificates [using the ingress-shim](../ingress/) alongside the [`ingress-gce`](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress) ingress controller, `ingress-gce` [required](https://github.com/kubernetes/ingress-gce/pull/388) a temporary certificate is to be present while waiting for the issuance of a signed certificate. Note that this issue was solved [in 1.10.7-gke.2](https://github.com/jetstack/cert-manager/issues/606#issuecomment-424397233).
> 
> To work around this, you had to add the following annotation to your Ingress objects:
>
> ```yaml
> cert-manager.io/issue-temporary-certificate": "true"
> ```
>
> That made sure that a temporary self-signed certificate is present in the `Secret`. The self-signed certificate is replaced with the properly signed certificate later on.

@munnerz I am correct in saying that this annotation is not needed anymore?

Reply:
I will resolve this conversation in order to move this PR forward. Please re-open it if you think I missed something.